### PR TITLE
[NOW] new, index, edit functions for Data_types

### PR DESCRIPTION
**Pull request by jonmchu**

### CODE REVIEW

File: app/views/data_types/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<h1>New Data Type</h1>
+  <%= render 'form' %>
```
Comment (SamLau95) [inline app/views/data_types/new.html.erb]: nit: No need to indent here.

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
 
+  get 'data_types/index'
```
Comment (SamLau95) [inline config/routes.rb]: Do you need this route? `resources :data_types` should give you the index route too. This looks like it was autogenerated.

File: app/controllers/data_types_controller.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    @data_type.save
+    redirect_to action:"index"
+  end
+
+  def update
+    @data_type = DataType.find(params[:id])
+
+    if @data_type.update(data_type_params)
+      redirect_to action:"index"
```
Comment (SamLau95) [inline app/controllers/data_types_controller.rb]: nit: Leave a space in between the colon and the quote. And in the `create` action too.

**Comment by jodreen** [comment]
add a link to the new page in the index, and there should be an extra column for delete

File: app/views/data_types/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  <table>
+    <tr>
+      <th>Name</th>
+      <th>Formula</th>
+    </tr>
+    <% @data_types.each do |data_type| %>
+        <tr>
+            <td><%= data_type.name %></td>
+            <td><%= data_type.formula %></td>
+            <td><%= link_to 'Edit', edit_data_type_path(data_type) %></td>
```
Comment (jodreen) [inline app/views/data_types/index.html.erb]: the indent should only be 2 spaces!

File: spec/factories/data_types.rb [hunk 5]
```diff
@@ -1,7 +1,6 @@
 FactoryGirl.define do
   factory :data_type do
-    formula "MyString"
-name "MyString"
+    formula "data_type_formula"
+    name "data_type_name"
```
Comment (jodreen) [inline spec/factories/data_types.rb]: :yellow_heart:

**Comment by kevintxwu** [comment]
I need this to merged tmmro to progress on the formula parser

**Comment by jodreen** [comment]
:full_moon_with_face:

**Pull request merged**